Alter onewire libs so other non-ibutton pins can be used as well

wdoekes opened the pull request:
# What's new

Changes so `onewire_host_alloc()` now takes a mandatory GpioPin. For iButton, this would be `&ibutton_gpio`. For other uses, the `&gpio_ext_pa7` (and other pins) can be used.

This changeset allows us to implement plugins/apps that speak the OneWire protocol over the top gpio pins.

Closes: #1596

# Verification 

- Someone needs to confirm that the iButton functionality still works. I have no such thing to test.
- To test the new behaviour, I hacked up a temperature sensor plugin where you can connect a DS18x20 temperature sensor and cycle through the pins using LEFT/RIGHT. Dirty version here: https://junk.devs.nu/2022/flipper/sensors_ds18x20_1.patch

![image](https://user-images.githubusercontent.com/1225014/185898200-955e2a75-c088-4838-be06-30f1cd4dbadd.png)

- In the image above, the DS sensor is connected to pin C0. The temperature reads 23.3 celcius.

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

wdoekes committed:
Alter onewire libs so other non-ibutton pins can be used as well

Changes so onewire_host_alloc() now takes a mandatory GpioPin. For
iButton, this would be &ibutton_gpio. For other uses, the &gpio_ext_pa7
(and other pins) can be used.

This changeset allows us to implement plugins/apps that speak the
OneWire protocol over the top gpio pins.

Closes: #1596

DrZlo13 commented on firmware/targets/f7/furi_hal/furi_hal_ibutton.c:
```diff
@@ -90,46 +91,13 @@ void furi_hal_ibutton_emulate_stop() {
 }
 
 void furi_hal_ibutton_start_drive() {
```
Please, remove furi_hal_ibutton_xx api, if not in use.

wdoekes replied:
Well.. it _was_ in use, but only by `furi_hal_rfid.c`. I replaced them with the onewire equivalents in an upcoming commit.

DrZlo13 commented on firmware/targets/furi_hal_include/furi_hal_onewire.h:
```diff
@@ -0,0 +1,79 @@
+ * Sets the pin to interrupt mode (EXTI interrupt on rise or fall), and sets it to float
+ * @param gpio
+ */
+void furi_hal_onewire_start_interrupt(const GpioPin* gpio);
```
Please comment which EXTI (which pin) can be used as an interrupt.
V9 table from https://docs.google.com/spreadsheets/d/1qAPFTwvMwbTp1hTAtZHlO6jpl9o-fXg49rPIQ1o7JiQ/edit?usp=sharing
(0, 1, 4, 5, 7, 8, 14).

wdoekes replied:
I'd love to add this, but I'm not sure what you mean. I'm not that initimate with electronics.

Are you saying all pins on A..H are interrupt-capable if they're on EXTI{0,1,4,5,7,8,14} and all others are not?

I.e.:
```c
/*
 * Sets the pin to interrupt mode (EXTI interrupt on rise or fall), and sets it to float
 * The following pins are valid:
 * gpio_infrared_rx, gpio_display_rst_n, gpio_ext_pc0, gpio_subghz_cs (EXTI0),
 * gpio_cc1101_g0, gpio_display_di, gpio_ext_pc1, gpio_spi_d_sck (EXTI1),
 * ... etc ...
```
?

I have no idea what I'm typing here, so I could be off by a mile :shrug:

wdoekes replied:
ping @DrZlo13 : any hint is appreciated :)

wdoekes committed:
Remove leftover furi_hal_ibutton_(start_drive|stop|pin_low)

Replace the calls with the new furi_hal_onewire_xx equivalents

Change request by @DrZlo13.

skotopes commented:
Please feel free to reopen after fixing all requested changes.

wdoekes commented:
Not fair @skotopes . I've asked for clarification, but not gotten any.

wdoekes commented:
Looks like you added DS18B20 support in 7a3a1aaf0db35501b6ee5891ad039605285f2fea as well :ok_hand:
